vdr/vdr.go

```diff
@@ -127,7 +133,22 @@ func (r VDR) Resolve(id did.DID, metadata *types.ResolveMetadata) (*did.Document
 
 // Update updates a DID Document based on the DID and current hash
 func (r VDR) Update(id did.DID, current hash.SHA256Hash, next did.Document, metadata *types.DocumentMetadata) error {
-	return r.store.Update(id, current, next, metadata)
+	resolverMetada := &types.ResolveMetadata{
```

Comment:
If I am correct we now first create the (updated) document on the store and then validate it when we receive it again. I'd like that we (also) validate it before storing it on the network (probably using the same validation rules) because that (hopefully) keeps the garbage on the network caused by errors (human/software) in updates to a minimum.

Reply:
should be single line of code, right.....

Reply:
Will add a TODO for this.